l1-contracts/test/Inbox.t.sol

```diff
@@ -210,6 +214,25 @@ contract InboxTest is Test {
     inbox.batchConsume(entryKeys, feeCollector);
   }
 
+  function testRevertIfConsumingFromWrongRollup() public {
+    address wrongRollup = address(0xbeeffeed);
+    uint256 wrongVersion = registry.upgrade(wrongRollup, address(inbox), address(0x0));
+
+    DataStructures.L1ToL2Msg memory message = _fakeMessage();
+    address feeCollector = address(0x1);
+    bytes32 entryKey = inbox.sendL2Message{value: message.fee}(
+      message.recipient, message.deadline, message.content, message.secretHash
+    );
+    bytes32[] memory entryKeys = new bytes32[](1);
+    entryKeys[0] = entryKey;
+
+    vm.prank(wrongRollup);
```

Comment:
can you add a comment which says that "this fails because the rollup version is 2 but we have sent a message for 1"?

Reply:
![](https://media.tenor.com/RZwzwW7YcwgAAAAd/captain-america-old-man.gif)

The test is named `testRevertIfConsumingFromWrongRollup` and 
```solidity
    vm.expectRevert(
      abi.encodeWithSelector(Errors.Inbox__InvalidVersion.selector, version, wrongVersion)
    );
```
makes the explicit check one line later than this.